### PR TITLE
Add cli support for Apache Kafka native ACLs

### PR DESCRIPTION
**Pull request by biggusdonzus**
Add the following `service` subcomands `kafka-acl-list` `kafka-acl-add` `kafka-acl-delete`.

### CODE REVIEW

File: aiven/client/cli.py [hunk 1]
```diff
@@ -6176,6 +6176,104 @@ def service__alloydbomni__google_cloud_private_key__delete(self) -> None:
...
+            "a name of '*' matches all resources of the selected type"
+        ),
+        required=True,
+    )
+    @arg(
+        "--resource-type",
+        help="Topic resource type to which ACLs should be added",
+        required=False,
+        choices=["Any", "Topic", "Group", "Cluster", "TransactionalId", "DelegationToken"],
+    )
```
Comment (tvainika) [inline aiven/client/cli.py:6214]: Just for user convenience, should we model command line args more similar to kafka-acls.sh?

It would mean that instead of simply giving resource-name as parameter, usage would be more like
`[--cluster | --group group | --topic topic | --transactional-id transactional-id ] `
where only one of these is given.

Also for cluster type name must be always `kafka-cluster`.

Also we could skip DelegationToken here because there is currently no support for those in aiven services.

File: aiven/client/cli.py [hunk 2]
```diff
@@ -6176,6 +6176,104 @@ def service__alloydbomni__google_cloud_private_key__delete(self) -> None:
...
+            operation=self.args.operation,
+            permission_type=self.args.permission_type,
+        )
+        print(response["message"])
+
+    @arg.project
+    @arg.service_name
+    @arg.json
+    def service__kafka_acl_list(self) -> None:
+        """List Kafka native ACL entries"""
```
Comment (tvainika) [inline aiven/client/cli.py:6258]:
```suggestion
        """List Kafka-native ACL entries"""
```

File: aiven/client/cli.py [hunk 3]
```diff
@@ -6176,6 +6176,104 @@ def service__alloydbomni__google_cloud_private_key__delete(self) -> None:
...
+        )
+        acls = response.get("kafka_acl", [])
+        layout = ["id", "principal", "operation", "resourceName", "resourceType", "patternType", "permissionType"]
+        self.print_response(acls, json=self.args.json, table_layout=layout)
+
+    @arg.project
+    @arg.service_name
+    @arg("acl_id", help="ID of the ACL entry to delete")
+    def service__kafka_acl_delete(self) -> None:
+        """Delete a Kafka ACL entry"""
```
Comment (tvainika) [inline aiven/client/cli.py:6271]:
```suggestion
        """Delete a Kafka-native ACL entry"""
```

File: aiven/client/cli.py [hunk 4]
```diff
@@ -6176,6 +6176,104 @@ def service__alloydbomni__google_cloud_private_key__delete(self) -> None:
...
+        required=False,
+        default="*",
+    )
+    @arg(
+        "--principal",
+        help="The principal for the ACLs, must be in the form principalType:name",
+        required=True,
+    )
+    def service__kafka_acl_add(self) -> None:
+        """Add a Kafka native ACL entry"""
```
Comment (tvainika) [inline aiven/client/cli.py:6240]:
```suggestion
        """Add a Kafka-native ACL entry"""
```

File: aiven/client/cli.py [hunk 5]
```diff
@@ -6176,6 +6176,104 @@ def service__alloydbomni__google_cloud_private_key__delete(self) -> None:
...
+    )
+    @arg(
+        "--permission-type",
+        help="The type of the resource pattern",
+        required=True,
+        choices=["ALLOW", "DENY"],
+    )
+    @arg(
+        "--host",
+        help="The host for the ACLs, a value of '*' matched all hosts",
```
Comment (tvainika) [inline aiven/client/cli.py:6230]:
```suggestion
        help="The host for the ACLs, a value of '*' matches all hosts",
```

File: aiven/client/cli.py [hunk 6]
```diff
@@ -6176,6 +6176,104 @@ def service__alloydbomni__google_cloud_private_key__delete(self) -> None:
...
+        help="The type of the resource pattern",
+        required=False,
+        choices=["LITERAL", "PREFIXED"],
+        default="LITERAL",
+    )
+    @arg(
+        "--permission-type",
+        help="The type of the resource pattern",
+        required=True,
+        choices=["ALLOW", "DENY"],
```
Comment (tvainika) [inline aiven/client/cli.py:6226]: More convenient if Allow is default?

**Review by tvainika — COMMENTED** [review]
In addition existing Aiven ACLs for Kafka need doc polishing to make it clear what is where (use `avn service --help` to see those existing commands)

File: aiven/client/cli.py [hunk 7]
```diff
@@ -6176,6 +6176,104 @@ def service__alloydbomni__google_cloud_private_key__delete(self) -> None:
...
+    @arg.service_name
+    @arg.json
+    def service__kafka_acl_list(self) -> None:
+        """List Kafka native ACL entries"""
+        response = self.client.service_kafka_native_acl_list(
+            project=self.get_project(),
+            service=self.args.service_name,
+        )
+        acls = response.get("kafka_acl", [])
+        layout = ["id", "principal", "operation", "resourceName", "resourceType", "patternType", "permissionType"]
```
Comment (tvainika) [inline aiven/client/cli.py:6264]: This is missing host. For convenience it could be so that host field is shown only if any host field is not `*`.

Also when I tested this first it printed nothing with zero kafka-native acls. Maybe somehow print always something even if no kafka-native acls are set.

Comment (tvainika) [inline aiven/client/cli.py:6264]: For readability resrouce type should be first (left side) of resource name

File: aiven/client/cli.py (hunk 6 above)
Comment (biggusdonzus) [reply aiven/client/cli.py:6226]: Going with this i modified this argument to be `--deny`, if not specified it is allow.

File: aiven/client/cli.py (hunk 7 above)
Comment (biggusdonzus) [reply aiven/client/cli.py]: :+1: for the Host, about the print with zero acls i'll think about it

Comment (biggusdonzus) [reply aiven/client/cli.py]: Now i've aligned this with the UX thread

File: aiven/client/cli.py (hunk 1 above)
Comment (tvainika) [reply aiven/client/cli.py]: This is argparse based so is it possible to add mutually exclusive group as in https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group ?

**Review by tvainika — APPROVED** [review]
